### PR TITLE
Fix speed benchmarks

### PR DESCRIPTION
**Pull request by not522**
## Motivation
This PR will fix the failure of speed benchmarks due to the change of `STORAGE_MODES` (#3204).

## Description of the changes
- Fix speed benchmarks
- Add `--strict` option of `asv` to fail the CI when some cases failing (https://github.com/airspeed-velocity/asv/pull/865)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def get_extras_require() -> Dict[str, List[str]]:
             "fastai",
         ],
         "benchmark": [
-            "asv",
+            "asv>=0.5.0",
```
Comment (not522) [inline setup.py:170]: Note: `--strict` was introduced in asv v0.5.0

**Comment by HideakiImamura** [comment]
@himkt Could you review this PR?

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Comment by himkt** [comment]
Confirmed that the job failed on strict mode (https://github.com/himkt/optuna-test-asv/runs/5563416229?check_suite_focus=true) and worked with patch (https://github.com/himkt/optuna-test-asv/runs/5563528557?check_suite_focus=true).

**Review by himkt — APPROVED** [review]
Thank you for the fix, LGTM!

**Pull request merged**